plugins/inputs/solrmetrics/solrmetrics.go

```diff
@@ -0,0 +1,413 @@
+			URLPrs["prefixes"] = url.Values{}
+			URLPrs["prefixes"].Set("prefix", strings.Join(s.Prefixes, ","))
+		}
+		// URL for a server
+		strURL := fmt.Sprintf("%s/solr%s", server, metricPath)
+		serverURL, err := url.Parse(strURL)
+		if err != nil {
+			acc.AddError(err)
+			continue
+		}
+		// Add common parameters and set parameters for the server URL
+		for _, v := range URLPrs {
+			v.Set("compact", compact)
+			v.Set("wt", wt)
+			serverURL.RawQuery = v.Encode()
+			URLs = append(URLs, *serverURL)
+		}
+
+	}
+	return URLs, nil
+}
+
+func (s *Solr) createHTTPClient() *http.Client {
+	// Set default value if the http timeout wasn't set in config
+	if s.Httptimeout == 0 {
+		s.Httptimeout = dftHTTPTimeout
+	}
+	tr := &http.Transport{
+		ResponseHeaderTimeout: s.Httptimeout * time.Second,
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   s.Httptimeout * time.Second,
+	}
+	return client
+}
+
+func (s *Solr) gatherData(req *http.Request, m interface{}) error {
+	r, err := s.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer r.Body.Close()
+	if r.StatusCode != http.StatusOK {
+		return fmt.Errorf("solr: API responded with status-code %d, expected %d, url %s",
+			r.StatusCode, http.StatusOK, req.URL)
+	}
+	if err = json.NewDecoder(r.Body).Decode(m); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateReqs creates request for the each Solr node
+func (s *Solr) CreateReqs(acc telegraf.Accumulator) ([]*http.Request, error) {
+
+	urls, err := s.metricsURLs(acc)
+	if urls == nil {
+		return nil, fmt.Errorf("empty list of URLs to request the metrics. %s", err)
+	}
+
+	var reqs []*http.Request
+	for _, url := range urls {
+		req, reqErr := http.NewRequest(http.MethodGet, url.String(), nil)
+		if reqErr != nil {
+			acc.AddError(err)
+			continue
+		}
+		if s.Username != "" {
+			req.SetBasicAuth(s.Username, s.Password)
+		}
+		req.Header.Set("User-Agent", "Telegraf/"+internal.Version())
+		reqs = append(reqs, req)
+	}
+	return reqs, nil
+}
+
+func (s *Solr) gatherServerMetrics(req *http.Request, acc telegraf.Accumulator) error {
+	mr := &MetricResponse{}
+	err := s.gatherData(req, mr)
+	if err != nil {
+		return err
+	}
+	metrics := parseMetrics(mr.Metrics, acc)
+	if len(metrics) == 0 {
+		return fmt.Errorf("no Solr metrics handled")
+	}
+	// Default tags
+	tags := make(Tags)
+	tags["port"] = req.URL.Port()
+	series := metricsToSeries(metrics, tags, acc)
+	if len(series) == 0 {
+		return fmt.Errorf("no metrics")
+	}
+	for _, serie := range series {
+		acc.AddFields(serie.mesName, serie.fields, serie.tags, serie.timestamp)
+	}
+
+	return nil
+}
+
+// Gather reads the stats from Solr and writes it to the
+// Accumulator.
+func (s *Solr) Gather(acc telegraf.Accumulator) (err error) {
+	if s.client == nil {
+		client := s.createHTTPClient()
+		s.client = client
+	}
+	if s.reqs == nil {
+		if s.reqs, err = s.CreateReqs(acc); err != nil {
+			acc.AddError(err)
+		}
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(s.reqs))
```

Comment:
Please move this into the for loop adding `1` during each execution of the loop.